### DISCUSSION TITLE
Adding a custom value converter to all properties of a certain type

### DISCUSSION THREAD

**Discussion by moxplod** [alirezanet/Gridify#213] [General]
Hi @alirezanet following up on #211 

I want to add a custom value converter in my code for EACH datetime property in the mappings.  

I have 2 thoughts here:

1. How can I iterate over each of these IGMaps and determine if they are a DateTime property or not? Should we add FromType in IGMap so that this can be inferred easily?

2. Or is it a better approach to create a Global configuration similar to CustomOperators and create CustomValueConvertors - then provide a Type and the convertor method. Something like a Dictionary<Type, Func<string, object>>

```c#
    private static IGridifyMapper<T> GridifyEntityMapper()
    {
        var mapper = new GridifyMapper<T>().GenerateMappings(2);
       // This is a hack to look at names - need a better way to find all DateTime and DateTime? properties.
       // Especially with all the nestings
        var dates = mapper.GetCurrentMaps().Where(m => m.From.Contains("date")).ToList();
        dates.ForEach(m => m.Convertor = input => Strings.ParseRelativeDate(input));
        return mapper;
    }
```

**Comment by alirezanet** [comment]
Hi @moxplod,
I think the better way of doing this is to have something like this:

```csharp
   public IEnumerable<IGMap<T>> GetCurrentMapsByType(HashSet<Type> targetTypes)
   {
      foreach (var map in _mappings)
      {
         if (map.To.Body is UnaryExpression unaryExpression) // this is needed for unboxing dateTimes
         {
            if (targetTypes.Contains(unaryExpression.Operand.Type))
            {
               yield return map;
            }
         }
         // TODO: need to implement other types that aren't unary expression
      }
   }
```

usage: 
```csharp
   [Fact]
   public void GetMappingByType()
   {
      var gm = new GridifyMapper<TestClass>(true);

      var dates = gm.GetCurrentMapsByType([typeof(DateTime), typeof(DateTime?)]);

      Assert.Single(dates);
   }
```

if we want to add this method to GridifyMapper we should also add another generic overload for this method. (`gm.GetCurrentMapsByType<DateTime>()`)
then you can use it to update the convertors

**Reply by moxplod** [reply]
This is a great approach. I will go ahead and build it this way and send a PR once ready,

**Comment by moxplod** [comment]
Added PR - https://github.com/alirezanet/Gridify/pull/214
